lib/rewards/server/subscribers/rewards-subscriber.js

```diff
@@ -0,0 +1,190 @@
+const { sc, Logger } = require('@reldens/utils');
+const { Modifier } = require('@reldens/modifiers');
+const { Reward } = require('../models/reward');
+
+class RewardsSubscriber
+{
+    static async giveRewards(playerSchema, targetObject, eventsManager)
+    {
+        let rewards = sc.get(targetObject, 'rewards', false);
+
+        let winningRewards = this.getWinningRewards(rewards);
+        if (0 === winningRewards.length) {
+            return false;
+        }
+        let repositories = this.getRepositories(targetObject);
+
+        let itemRewards = [];
+        for (let winningReward of winningRewards) {
+            let giveRewardResults = await this.giveReward(winningReward, repositories, playerSchema, async (item) => {
+                itemRewards.push({
+                    item,
+                    reward: winningReward
+                });
+            });
+            if (0 < giveRewardResults.length) {
+                Logger.error('There were error(s) on the reward ' + winningReward.id +
```

Comment:
I'm going to create something like sprintf from PHP into the shortcuts to avoid having all the concatenations we already have so we can use it like: `sc.sprintf('string with %s %s', paramForS1, paramForS2);`

Reply:
We can just use string template ``, in my opinion it's look better and it's make easier to read the code.